WangscapeTest/TestCalculatorLinear.cpp

```diff
@@ -0,0 +1,93 @@
+#include <gtest/gtest.h>
+#include <random>
+#include <numeric>
+#include <algorithm>
+
+#include <tilegen/alpha/CalculatorLinear.h>
+
+class TestCalculatorLinear : public ::testing::Test
+{
+public:
+    tilegen::alpha::CalculatorLinear cl;
+    std::mt19937 rng;
+    TestCalculatorLinear()
+    {
+        std::random_device rd;
+        rng.seed(rd());
+    }
+    ptrdiff_t validateResults(const tilegen::alpha::CalculatorBase::Weights& weights)
+    {
+        cl.updateAlphas(weights);
+        const auto& alphas = cl.getAlphas();
+        int total_alpha = std::accumulate(alphas.cbegin(), alphas.cend(), 0);
+        EXPECT_EQ(255, total_alpha);
+        double total_weight = std::accumulate(weights.cbegin(), weights.cend(), 0.);
+        for (size_t i = 0; i < 4; i++)
+        {
+            double alpha_fraction = double(alphas[i]) / 255;
+            double weight_fraction = double(weights[i]) / total_weight;
+            EXPECT_NEAR(alpha_fraction, weight_fraction, 1.5 / 255);
```

Comment:
Alpha values returned by the `alpha::CalculatorBase` are quantised as integers from 0 to 255, producing an error up to `0.5/255`. When the initial division of alpha points has a remainder, [up to 1 point can be added to each corner](https://github.com/serin-delaunay/Wangscape/blob/master/Wangscape/tilegen/alpha/CalculatorLinear.cpp#L57), increasing the maximum error to `1.5/255`. I'll add a comment explaining that.